pkg/virtctl/convert/guess.go

```diff
@@ -0,0 +1,36 @@
+package convert
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"k8s.io/client-go/pkg/util/yaml"
+	"unicode"
+)
+
+func GuessStreamType(source io.Reader, size int) (io.Reader, Type) {
+	if r, _, isXML := guessXMLStream(source, size); isXML {
+		return r, XML
+	} else if r, _, isJSON := yaml.GuessJSONStream(r, size); isJSON {
+		return r, JSON
+	} else {
+		return r, YAML
+	}
+}
+
+func guessXMLStream(r io.Reader, size int) (io.Reader, []byte, bool) {
+	buffer := bufio.NewReaderSize(r, size)
+	b, _ := buffer.Peek(size)
+	return buffer, b, hasXMLPrefix(b)
+}
+
+var xmlPrefix = []byte("<")
```

Comment:
I think it is good enough. Everything more than that would involve parsing the whole xml (basically a validation then, which will be done by the decoder later on)